From 810d71a7fb0c496b6fda8d4ade03ab6139b32afc Mon Sep 17 00:00:00 2001
From: Evan Huus <eapache@gmail.com>
Date: Thu, 6 Jun 2013 02:53:36 +0000
Subject: [PATCH 8/8] From myself and Julian Cable via (and fixing)
 https://bugs.wireshark.org/bugzilla/show_bug.cgi?id=8717

Don't add a DCP-ETSI fragment for reassembly if the length is wrong.

svn path=/trunk/; revision=49802
---
 epan/dissectors/packet-dcp-etsi.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/epan/dissectors/packet-dcp-etsi.c b/epan/dissectors/packet-dcp-etsi.c
index a90407c..b902255 100644
--- a/epan/dissectors/packet-dcp-etsi.c
+++ b/epan/dissectors/packet-dcp-etsi.c
@@ -519,9 +519,10 @@ dissect_pft(tvbuff_t * tvb, packet_info * pinfo, proto_tree * tree)
       if(li)
         proto_item_append_text(li, " (length error (%d))", real_len);
     }
-    if (real_len)
+    else {
       next_tvb = dissect_pft_fragmented(tvb, pinfo, pft_tree, findex, fcount,
                                         seq, offset, real_len, fec, rsk, rsz);
+    }
     pinfo->fragmented = save_fragmented;
   } else {
     next_tvb = tvb_new_subset_remaining (tvb, offset);
-- 
1.7.10.4

